lib/nexmo_developer/app/services/translator/smartling_downloader.rb

```diff
@@ -0,0 +1,49 @@
+module Translator
+  class SmartlingDownloader
+    include ::Translator::Utils
+
+    attr_reader :paths
+
+    def initialize(paths: [])
+      @paths = paths || file_paths
+    end
+
+    def call
+      raise ArgumentError, "The 'paths' parameter cannot be empty" unless @paths
+
+      @paths.map do |path|
+        locales = get_file_status(path: path)
+        locales.map { |locale| download_file(locale: locale, path: path) }
+      end
+    end
+
+    def file_paths
+      ::Translator::Smartling::ApiRequestsGenerator.file_paths
+    end
+
+    def get_file_status(path:)
+      ::Translator::Smartling::ApiRequestsGenerator.get_file_status(
+        path: path
+      )
+    end
+
+    def download_file(locale:, path:)
+      doc = ::Translator::Smartling::ApiRequestsGenerator.download_file(
+        locale: locale,
+        path: path
+      )
```

Comment:
This will only be called on file uris that Smartling returns to us that are published and ready to be downloaded. The error possibility is slim to none.

Reply:
The thing is that the API could return a 500/4xx or it could even timeout, so we'll get `nil`  here.